fs2-aws-s3/src/main/scala/fs2/aws/s3/S3.scala

```diff
@@ -213,7 +213,7 @@ object S3 {
 
         def checksumPart(chunk: Chunk[Byte]): PartDigest = {
           val md = createMD5
-          chunk.foreach(byte => md.update(byte))
+          md.update(chunk.toArray)
```

Comment:
@semenodm so one thing in our apps is I've tried to not create new shortly lived arrays, this will create a short lived array of 5MB which puts memory pressure on the app.  This is the reason I had used `chunk.foreach` to prevent this allocation and memory pressure.